### PR TITLE
Setup scraper for verenigingen harvester

### PR DESCRIPTION
**Pull request by oscartison**
This micro-service will be used to scrape an endpoint of an external provider. It should check an endpoint  https://publiek.verenigingen.staging-vlaanderen.be/v1/verenigingen/zoeken)  where a list of associations is returned and create a json file with the every association that should be checked (https://publiek.verenigingen.staging-vlaanderen.be/v1/verenigingen/zoeken/V001001, https://publiek.verenigingen.staging-vlaanderen.be/v1/verenigingen/zoeken/V001002,.... )

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Copyright 2023 REDPENCIL BV (redpencil.io)
```
Comment (nbittich) [inline LICENSE:1]: should be Vlaamse Overheid

File: lblod/pipeline.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+from lblod.file import STORAGE_PATH, construct_insert_file_query
+from lblod.harvester import collection_has_collected_files, create_results_container
+from lblod.job import update_task_status
+from sudo_query import update_sudo
+from helpers import logger
+
+
+def get_item(rdo, task):
+    # Get all possible urls
+    json_urls = {"urls": ["https://verenigingen.oscart-dev.s.redhost.be/json-ld?page=1",
```
Comment (nbittich) [inline lblod/pipeline.py:20]: I'm a bit confused here. those url should come from the task created by either an user or a cron scheduler, it seems they are hardcoded.
You don't seem to use rdo and task in that function

File: lblod/pipeline.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+def process_item(content, rdo):
+    if not os.path.exists(STORAGE_PATH):
+        os.mkdir(STORAGE_PATH)
+
+    _uuid = str(uuid.uuid4())
+    json_file_name = f"{_uuid}.json.gz"
+    json_file_path = os.path.join(STORAGE_PATH, json_file_name)
+
+    with gzip.open(json_file_path, "wt", encoding='utf-8') as f:
+        f.write(content)
```
Comment (nbittich) [inline lblod/pipeline.py:35]: the rest of the harvester stack requires N-TRIPLE format to work properly. 
There is a need for either an extra  to convert  from json-ld to N-TRIPLE I think

**Review by nbittich — COMMENTED** [review]
I've added my comments, it seems a list of urls is hardcoded instead of extracted from the task (the initial remote data object).

Instead, it should look at the first url provided by the user (from the rdo), download the json-ld from that page, check if in the json-ld there are other url to harvest, if so, create new remote data objects for each of them and continue until everything has been downloaded. I didn't see that so far in the code, but I could be wrong. Please let me know if / where I missed that part.

File: lblod/pipeline.py (hunk 2 above)
Comment (oscartison) [reply lblod/pipeline.py:21]: Yes, so for the moment we do not have access to the API that we should harvest but from what the provider told us, with url that the user will input, we will get a response that looks like this 
```
{[{vcode: V100100}, {vcode: V100101}]}
```
but we want to scrape the detail page of that API, so URL/detail/V100100 and URL/detail/V100101 . As we don't have the API yet, we just mocked this.

File: lblod/pipeline.py (hunk 3 above)
Comment (oscartison) [reply lblod/pipeline.py:36]: We also added a second service that does the conversion: https://github.com/lblod/harvesting-verenigingen-import-service

**Pull request merged**